internal/mempool/mempool.go

```diff
@@ -364,7 +364,7 @@ func (txmp *TxMempool) RemoveTxByKey(txKey types.TxKey) error {
 
 	// remove the committed transaction from the transaction store and indexes
 	if wtx := txmp.txStore.GetTxByHash(txKey); wtx != nil {
-		txmp.removeTx(wtx, false, true)
+		txmp.removeTx(wtx, false, true, true)
```

Comment:
(once we have so many flags, it might make sense to split these out as named methods)

Reply:
yeah makes sense. Or group them into a `RemoveTxOptions` struct that has named fields?